python/whylogs/core/constraints/factories/distribution_metrics.py

```diff
@@ -15,19 +15,14 @@ def greater_than_number(column_name: str, number: Union[float, int], skip_missin
         reference value for applying the constraint
     skip_missing: bool
         If skip_missing is True, missing distribution metrics will make the check pass.
-        If False, the check will fail on missing metrics
+        If False, the check will fail on missing metrics, such as on an empty dataset
     """
 
-    def is_greater(metric: DistributionMetric) -> bool:
-        if not metric.kll.value.is_empty():
-            return metric.min >= number
-        else:
-            return True if skip_missing else False
-
     constraint = MetricConstraint(
         name=f"{column_name} greater than number {number}",
-        condition=is_greater,
+        condition=lambda x: x.min > number,
```

Comment:
why has this changed back to lambda?

Reply:
previously, the function needed to make additional validation with regards to missing distribution metrics. now this is covered by `require_column_existence`. This led to a very simple one-line function.

Since it was a single line, I decided to make it a lambda. But now I just recalled a conversation where we agreed that we should prefer functions to lambdas. I'm not sure what the reason was, though. Is it because it's easier to debug? any other reason?

Reply:
The same comment applies to all the other cases below

Reply:
I think you could do `Require("min").greater_than(number)`.  Lambdas are often easier to code, but cannot be [de]serialized.

Reply:
replaced lambda to Predicate in latest commit